src/services/podcast.ts

```diff
@@ -51,28 +51,29 @@ export const getPodcasts = async (query: any = {}) => {
 export const getSubscribedPodcasts = async (subscribedPodcastIds: [string]) => {
   const addByRSSPodcasts = await getAddByRSSPodcastsLocally()
 
-  if (subscribedPodcastIds.length < 1 && addByRSSPodcasts.length < 1) return [[], 0]
-
-  if (subscribedPodcastIds.length < 1 && addByRSSPodcasts.length > 0) {
-    const combinedPodcasts = await combineWithAddByRSSPodcasts()
-    return [combinedPodcasts, combinedPodcasts.length]
-  }
-
   const query = {
     podcastIds: subscribedPodcastIds,
     sort: PV.Filters._alphabeticalKey,
     maxResults: true
   }
   const isConnected = await hasValidNetworkConnection()
 
+  if (subscribedPodcastIds.length < 1 && addByRSSPodcasts.length < 1) return [[], 0]
+
+  if (subscribedPodcastIds.length < 1 && addByRSSPodcasts.length > 0) {
+    if (isConnected) await parseAllAddByRSSPodcasts()
```

Comment:
Does this need to be an await? Is the rest of this functions code dependent on this completing?